scripts/dataset_processing/tts/hui_acg/get_data.py

```diff
@@ -19,156 +19,296 @@
     if not destination_path.exists():
-        tmp_file_path = destination_path.with_suffix('.tmp')
-        urllib.request.urlretrieve(source_url, filename=str(tmp_file_path))
+        logging.info(f"Downloading data: {source_url} --> {destination_path}")
+        tmp_file_path = destination_path.with_suffix(".tmp")
+        urllib.request.urlretrieve(source_url, filename=tmp_file_path)
         tmp_file_path.rename(destination_path)
+    else:
+        logging.info(f"Skipped downloading data because it exists: {destination_path}")
 
 
 def __extract_file(filepath, data_dir):
-    try:
-        shutil.unpack_archive(filepath, data_dir)
-    except Exception:
-        print(f"Error while extracting {filepath}. Already extracted?")
-
-
-def __process_data(dataset_path, stat_path, min_duration, max_duration, val_size, test_size, seed_for_ds_split):
-    # Create normalizer
-    text_normalizer = Normalizer(
-        lang="de", input_case="cased", overwrite_cache=True, cache_dir=str(dataset_path / "cache_dir"),
-    )
-    text_normalizer_call_kwargs = {"punct_pre_process": True, "punct_post_process": True}
-    normalizer_call = lambda x: text_normalizer.normalize(x, **text_normalizer_call_kwargs)
-
+    logging.info(f"Unzipping data: {filepath} --> {data_dir}")
+    shutil.unpack_archive(filepath, data_dir)
+    logging.info(f"Unzipping data is complete: {filepath}.")
+
+
+def __save_json(json_file, dict_list):
+    logging.info(f"Saving JSON split to {json_file}.")
+    with open(json_file, "w") as f:
+        for d in dict_list:
+            f.write(json.dumps(d) + "\n")
+
+
+def __process_data(
+        dataset_path, stat_path_root, speaker_id, min_duration, max_duration, val_size, test_size, seed_for_ds_split,
+):
+    logging.info(f"Preparing JSON split for speaker {speaker_id}.")
+    # parse statistic.txt
+    stat_path = stat_path_root / "statistic.txt"
+    with open(stat_path, 'r') as fstat:
+        lines = fstat.readlines()
+        num_utts = int(lines[4].strip().split()[-1])
+        hours = round(float(lines[9].strip().split()[-1]) / 3600.0, 2)
+
+    # parse overview.csv to generate JSON splits.
+    overview_path = stat_path_root / "overview.csv"
     entries = []
-    with open(stat_path) as f:
+    with open(overview_path, 'r') as foverview:
         # Let's skip the header
-        f.readline()
-        for line in tqdm(f):
+        foverview.readline()
+        for line in tqdm(foverview):
             file_stem, duration, *_, text = line.strip().split("|")
             duration = float(duration)
 
-            # file_stem -> dir_name (e.g. maerchen_01_f000051 -> maerchen, ber_psychoanalyse_01_f000046 -> ber_psychoanalyse)
+            # file_stem -> dir_name (e.g. maerchen_01_f000051 -> maerchen)
             dir_name = "_".join(file_stem.split("_")[:-2])
             audio_path = dataset_path / dir_name / "wavs" / f"{file_stem}.wav"
 
             if min_duration <= duration <= max_duration:
```

Comment:
This would drop some datapoints from the original dataset, do we want to do this or give the user the option to do so in the get_data.py file?

Reply:
Yes. That's designed on purpose. We provided min and max duration arguments. Please check Line 60 and 61.